src/lib/coda_networking/coda_networking.ml

```diff
@@ -400,12 +419,17 @@ module Make (Inputs : Inputs_intf) = struct
     Logger.trace t.log !"Broadcasting %{sexp: Message.msg} over gossip net" x ;
     Linear_pipe.write_without_pushback (Gossip_net.broadcast t.gossip_net) x
 
-  let broadcast_state t x = broadcast t (New_state x)
+  let broadcast_from_me t content =
```

Comment:
@cmr.  a peer can potentially forge their `discovery_port` and `communication`. Are you fine with that?

Reply:
Yes, that's OK. We just can't allow them to forge their IP.